packages/flutter_tools/lib/src/project.dart

```diff
@@ -509,30 +509,33 @@ class AndroidProject extends FlutterProjectPlatform {
     if (plugin.existsSync()) {
       return false;
     }
-    final File appGradle = hostAppGradleRoot.childFile(
-        fileSystem.path.join('app', 'build.gradle'));
-    if (!appGradle.existsSync()) {
-      return false;
-    }
-    for (final String line in appGradle.readAsLinesSync()) {
-      final bool fileBasedApply = line.contains(RegExp(r'apply from: .*/flutter.gradle'));
-      final bool declarativeApply = line.contains('dev.flutter.flutter-gradle-plugin');
-      final bool managed = line.contains("def flutterPluginVersion = 'managed'");
-      if (fileBasedApply || declarativeApply || managed) {
-        return true;
+    try {
+      for (final String line in appGradleFile.readAsLinesSync()) {
+        final bool fileBasedApply = line.contains(RegExp(r'apply from: .*/flutter.gradle'));
+        final bool declarativeApply = line.contains('dev.flutter.flutter-gradle-plugin');
+        final bool managed = line.contains("def flutterPluginVersion = 'managed'");
+        if (fileBasedApply || declarativeApply || managed) {
+          return true;
+        }
       }
+    } on FileSystemException {
+      return false;
     }
     return false;
   }
 
   /// True, if the app project is using Kotlin.
   bool get isKotlin {
-    final File gradleFile = hostAppGradleRoot.childDirectory('app').childFile('build.gradle');
-    final bool imperativeMatch = firstMatchInFile(gradleFile, _imperativeKotlinPluginPattern) != null;
-    final bool declarativeMatch = firstMatchInFile(gradleFile, _declarativeKotlinPluginPattern) != null;
+    final bool imperativeMatch = firstMatchInFile(appGradleFile, _imperativeKotlinPluginPattern) != null;
+    final bool declarativeMatch = firstMatchInFile(appGradleFile, _declarativeKotlinPluginPattern) != null;
     return imperativeMatch || declarativeMatch;
   }
 
+  /// Gets the module-level build.gradle file.
+  /// See https://developer.android.com/build#module-level.
+  File get appGradleFile => hostAppGradleRoot.childDirectory('app')
```

Comment:
I cant find any documentation on this or anyone who knows it exists and I have found code that also assumes app is the folder and hardcodes the value. https://github.com/christopherfujino/flutter/blob/re-enable-packages-autoroller/packages/flutter_tools/lib/src/project.dart#L513

~Not all flutter projects have the sub folder named app. I think you need to check the configuration for what the name of the project folder should be. When I can find documentation or code for that feature I will link it here. ~

Reply:
Yes, this came up earlier in discussion and I should have mentioned explicitly on this PR - the reason I refactored to make us use a common getter here is that it makes it easy for us to update how we are getting this gradle file. 

I can include those changes in this PR if you want, but the assumption is made in many places and feels logically separate from this migrator, so my plan was to write the migrator in a way that doesn't add an additional hardcoded place so that is still an easy follow up.

Would you prefer this PR also fix that assumption in the places we make it?

Reply:
Resolving this per offline discussion that it isn't a blocker for this PR, and we aren't sure that it is currently possible for the folder to be anything but app.